client/clientservice/src/state_snapshot_service.cpp

```diff
@@ -251,7 +251,14 @@ Status StateSnapshotServiceImpl::GetRecentSnapshot(ServerContext* context,
 
   if (return_status.ok() && snapshot_response.data.has_value()) {
     response->set_snapshot_id(snapshot_response.data->snapshot_id);
-    response->set_event_group_id(snapshot_response.data->event_group_id);
+    switch (snapshot_response.data->blockchain_height_type) {
+      case concord::messages::BlockchainHeightType::EventGroupId:
+        response->set_event_group_id(snapshot_response.data->blockchain_height);
+        break;
+      case concord::messages::BlockchainHeightType::BlockId:
+        response->set_block_id(snapshot_response.data->blockchain_height);
+        break;
```

Comment:
nit: we can add default: with a LOG_FATAL() and ConcordAssert(false). This will help us identify the issue if we add any new height type. But lets take that in a later PR.

Reply:
I rely on the compiler not allowing us to compile without handling all enum variants. That is why I prefer to omit `default`:

```
/concord-bft/client/clientservice/src/state_snapshot_service.cpp:254:13: error: enumeration value 'BlockId' not handled in switch [-Werror,-Wswitch]
    switch (snapshot_response.data->blockchain_height_type) {
```